cpp/src/arrow/compute/kernels/scalar_string.cc

```diff
@@ -672,11 +678,64 @@ struct Utf8TitleTransform : public FunctionalCaseMappingTransform {
 template <typename Type>
 using Utf8Title = StringTransformExec<Type, Utf8TitleTransform>;
 
+struct Utf8NormalizeTransform : public FunctionalCaseMappingTransform {
+  using State = OptionsWrapper<Utf8NormalizeOptions>;
+
+  const Utf8NormalizeOptions* options;
+
+  explicit Utf8NormalizeTransform(const Utf8NormalizeOptions& options)
+      : options{&options} {}
+
+  int64_t MaxCodeunits(const uint8_t* input, int64_t ninputs,
+                       int64_t input_ncodeunits) override {
+    const auto option = GenerateUtf8NormalizeOption(options->method);
```

Comment:
Since this is a constant value that only depends on the given options, we can move the `GenerateUtf8NormalizeOption` to the `Utf8NormalizeOptions` and call it in the constructor, and add a `utf8proc_option_t utf8proc_form` data member to hold this value. Then, this value can be accessed in the kernel as `options_.utf8proc_form`.

Reply:
Thanks for Advice!

I'm wondering how to fix it.

If I fix according to the advice, It need to add code to api_scalar.h like this.
```
#ifdef ARROW_WITH_UTF8PROC
#include <utf8proc.h>
#endif
```

#ifdef ARROW_WITH_UTF8PROC
struct ARROW_EXPORT Utf8NormalizeOptions : public FunctionOptions {
   Utf8NormalizeOptions(utf8proc_option_t utf8proc_form) {
      ...
   }
}
#endif
``` 

But including utf8proc in the public header is not a good solution, right?

Could you give me a hint?